str/str_iterative_hipstr_runner.py

```diff
@@ -0,0 +1,143 @@
+@click.command()
+def main(
+    variant_catalog, dataset, external_wgs_ids, output_file_name
+):  # pylint: disable=missing-function-docstring
+    b = get_batch()
+    #Create HipSTR job 
+    hipstr_job = b.new_job(name=f'HipSTR running')
+    hipstr_job.image(HIPSTR_IMAGE)
+    hipstr_job.storage('375G')
+    hipstr_job.cpu(16)
+
+    hipstr_job.declare_resource_group(
+        hipstr_output={
+            'vcf.gz': '{root}.vcf.gz',
+            'viz.gz': '{root}.viz.gz',
+            'log.txt': '{root}.log.txt',
+        }
+    )
+    hipstr_job.cloudfuse(f'cpg-{dataset}-main', '/cramfuse')
+    ref_fasta = 'gs://cpg-common-main/references/hg38/v0/Homo_sapiens_assembly38.fasta'
+
+    #Extract CPG IDs from external sample IDs
+    external_id_to_cpg_id: dict[str, str] = SampleApi().get_sample_id_map_by_external(
+        dataset, list(external_wgs_ids)
+    )
+
+    analysis_query_model = AnalysisQueryModel(
+        sample_ids=list(external_id_to_cpg_id.values()),
+        projects=[dataset],
+        type=AnalysisType('cram'),
+        status=AnalysisStatus('completed'),
+        meta={'sequence_type': 'genome', 'source': 'nagim'},
+    )
+
+    crams_path = AnalysisApi().query_analyses(analysis_query_model)
+    cpg_sids_with_crams = set(sid for sids in crams_path for sid in sids['sample_ids'])
+    cpg_id_to_external_id = {
+        cpg_id: external_wgs_id
+        for external_wgs_id, cpg_id in external_id_to_cpg_id.items()
+    }
+    cpg_sids_without_crams = set(cpg_id_to_external_id.keys()) - cpg_sids_with_crams
+    if cpg_sids_without_crams:
+        external_wgs_sids_without_crams = ', '.join(
+            cpg_id_to_external_id[sid] for sid in cpg_sids_without_crams
+        )
+        logging.warning(
+            f'There were some samples without CRAMs: {external_wgs_sids_without_crams}'
+        )
+    #Create string containing paths based on /cramfuse
+    cramfuse_path = ''
+    for cram_obj in crams_path:
+        cpg_sample_id = cram_obj['sample_ids'][0]
+        if dataset == 'tob-wgs':
+            cramfuse_path += '/cramfuse/cram/nagim/' + cpg_sample_id + '.cram,'
+        else:
+            cramfuse_path += '/cramfuse/cram/' + cpg_sample_id + '.cram,'
+    cramfuse_path = crams_path[:-1] #removes the terminating comma 
+
+    #Read in HipSTR variant catalog
+    hipstr_regions = b.read_input(variant_catalog)
+
+    #Read in reference 
+    ref = b.read_input_group(
+        **dict(
+            base=ref_fasta,
+            fai=ref_fasta + '.fai',
+            dict=ref_fasta.replace('.fasta', '').replace('.fna', '').replace('.fa', '')
+            + '.dict',
+        )
+    )
+
+    hipstr_job.command(
+        f"""
+    HipSTR --bams {cramfuse_path} --fasta {ref.base} --regions {hipstr_regions} --str-vcf {hipstr_job.hipstr_output['vcf.gz']} --viz-out {hipstr_job.hipstr_output['viz.gz']} --log {hipstr_job.hipstr_output['log.txt']} --output-filters
```

Comment:
Nitpick, but I think it's cleaner to keep these on separate lines, and makes looking at the command in Hail Batch easier:

```suggestion
    HipSTR --bams {cramfuse_path} \\
        --fasta {ref.base} \\
        --regions {hipstr_regions} \\
        --str-vcf {hipstr_job.hipstr_output['vcf.gz']} \\
        --viz-out {hipstr_job.hipstr_output['viz.gz']} \\
        --log {hipstr_job.hipstr_output['log.txt']} \\
        --output-filters
```